processing/src/main/java/io/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -249,8 +250,8 @@ private ColumnAnalysis analyzeStringColumn(
     }
 
     if (analyzingSize()) {
-      final long start = storageAdapter.getMinTime().getMillis();
-      final long end = storageAdapter.getMaxTime().getMillis();
+      final DateTime start = storageAdapter.getMinTime();
```

Comment:
Those `start` and `end` variables are used in an `Interval` constructor below. If it accepts `long` (timestamps), it uses the default system time zone. If it accepts DateTime objects, it borrows the timeZone from the `start`.